### PR TITLE
[fix]: [CCM-13147]: BigQuery Exception if billing table does not exist

### PR DESCRIPTION
**Pull request by iamanmolm**
## Describe your changes
We get com.google.cloud.bigquery.BigQueryException when billing table for an account is not present but visibility feature is enabled and we are able to fetch the recommendation so putting try catch around it, so we dont stop recommendation to be visible in the recommendations page.
Since this query is only used for getting the utilization data.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49871)

### CODE REVIEW

**Comment by iamanmolm** [comment]
trigger uts

File: 374-ceng-graphql/src/main/java/io/harness/ccm/graphql/core/recommendation/AzureCpuUtilisationService.java [hunk 1]
```diff
@@ -75,6 +76,8 @@ public List<AzureVmUtilisationDTO> getAzureVmCpuUtilisationData(String vmId, Str
     } catch (InterruptedException e) {
       log.error("Failed to get Azure VM Cpu Utilisation data for Account:{}, {}", accountIdentifier, e);
       Thread.currentThread().interrupt();
+    } catch (BigQueryException e) {
+      log.error("Failed to get Azure VM Cpu Utilisation data for Account:{}, {}", accountIdentifier, e);
```
Comment (abhinavsingh-ccm) [inline 374-ceng-graphql/src/main/java/io/harness/ccm/graphql/core/recommendation/AzureCpuUtilisationService.java:80]: Do we really need to log complete exception stacktrace? If not, we can print exception message instead. `e.getMessage()`

**Review by abhinavsingh-ccm — APPROVED** [review]
1 small comment

File: 374-ceng-graphql/src/main/java/io/harness/ccm/graphql/core/recommendation/AzureCpuUtilisationService.java (hunk 1 above)
Comment (iamanmolm) [reply 374-ceng-graphql/src/main/java/io/harness/ccm/graphql/core/recommendation/AzureCpuUtilisationService.java:80]: When I looked at other places where we are logging big query exception we are logging the complete exception stacktrace.
ex: https://github.com/harness/harness-core/blob/develop/400-rest/src/main/java/io/harness/ccm/billing/bigquery/BigQueryServiceImpl.java#L91, https://github.com/harness/harness-core/blob/develop/batch-processing/service/src/main/java/io/harness/batch/processing/anomalydetection/service/impl/AnomalyDetectionBigQueryServiceImplNG.java#L128 or lot of other examples

Comment (abhinavsingh-ccm) [reply 374-ceng-graphql/src/main/java/io/harness/ccm/graphql/core/recommendation/AzureCpuUtilisationService.java:80]: it's good time to initiate cleaning.

**Review by abhinavsingh-ccm — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49871)* status: 0 of 1 approvals obtained.

**Review by abhinavsingh-ccm — APPROVED** [review]

**Pull request merged**